adapters/eea-direct.js

```diff
@@ -0,0 +1,265 @@
+      }
+      const data = parse(body, {delimiter: '\t'});
+      getCoordinates(data, source.country, cb);
+    });
+  };
+};
+
+// reduce metadata to list of objects with coordinates for
+const getCoordinates = (metadata, country, callback) => {
+  // filter for only country of interest's records
+  filter(metadata, (record, truth) => {
+    truth(record[0] === country);
+  }, (countryMetadata) => {
+    // map filtered records to be a list of objs w stationId/coordinates
+    map(countryMetadata, (record, done) => {
+      const station = {
+        stationId: record[5],
+        coordinates: {
+          latitude: parseFloat(record[15]),
+          longitude: parseFloat(record[14])
+        }
+      };
+      done(null, station);
+    }, (err, mappedRecords) => {
+      if (err) {
+        return callback(null, []);
+      }
+      callback(null, uniqBy(mappedRecords, 'stationId'));
+    });
+  });
+};
+
+// makes requests to get country's pollutant data.
+const makeTaskRequests = (source) => {
+  const pollutantRequests = acceptableParameters.map((pollutant) => {
+    pollutant = pollutant.toUpperCase();
+    return (done) => {
+      const url = source.url.replace('<pollutant>', pollutant);
+      request.get({
+        url: url
+      }, (err, res, body) => {
+        if (err || res.statusCode !== 200) {
+          return done(null, []);
+        }
+        done(null, parse(body).slice(1, -1));
+      });
+    };
+  });
+  return (done) => {
+    parallel(
+      pollutantRequests,
+      (err, response) => {
+        if (err) {
+          done(null, []);
+        }
+        done(null, [].concat.apply([], response));
+      }
+    );
+  };
+};
+
+// formats data to match openAQ standard
+const formatData = (data, source, cb) => {
+  const coordinates = data[0];
+  const records = data[1];
+  parallel([
+    (done) => {
+      map(records, (record, cb) => {
+        let measurement = {
+          coordinates: makeCoordinates(coordinates, record[11]),
+          parameter: record[5].toLowerCase(),
+          date: makeDate(record[16], record[4]),
+          value: parseInt(record[19]),
+          unit: record[record.length - 1],
+          attribution: [{
+            name: 'EEA',
+            url: source.sourceUrl
+          }],
+          averagingPeriod: {
+            unit: 'hours',
+            value: 1
+          }
+        };
+        // apply unit conversion to generated record
+        cb(null, convertUnits([measurement])[0]);
+      }, (err, mappedRecords) => {
+        if (err) {
+          return done(null, []);
+        }
+        done(null, mappedRecords);
+      });
+    },
+    (done) => {
+      // try to combine locations with coordinates
+      // by matching stationId or a spatial join.
+      // then return successful combos
+      request.get({url: stationsLink}, (err, res, locations) => {
+        if (err) {
+          return done(null, []);
+        }
+        locations = JSON.parse(locations);
```

Comment:
Wrap in a try/catch please. JSON.parse's going bad have totally destroyed me before. :sob: